cpp/src/io/parquet/page_data.cu

```diff
@@ -1127,7 +1119,7 @@ static __device__ bool setupLocalPageInfo(page_state_s *const s,
         s->page.skipped_leaf_values = s->first_row;
       }
 
-      s->input_value_count = s->page.skipped_values;
+      s->input_value_count = s->page.skipped_values > -1 ? s->page.skipped_values : 0;
```

Comment:
This was causing race condition where default `s->page.skipped_values` is `-1` which gets assigned to `s->input_value_count`  which is supposed to be `0`. Then while indexing with thread 0, -1 will be converted to 255, causing access of improper value [here](https://github.com/rapidsai/cudf/blob/d1e8fee20b1c25059d5d3a35e7bd862fa0d6c6d9/cpp/src/io/parquet/page_data.cu#L1225)

Reply:
https://github.com/rapidsai/cudf/blob/d1e8fee20b1c25059d5d3a35e7bd862fa0d6c6d9/cpp/src/io/parquet/page_data.cu#L1225